### ISSUE TITLE
Use `pyupgrade`

### ISSUE THREAD

**Issue by forsyth2** [E3SM-Project/zppy#218]
Make use of https://github.com/asottile/pyupgrade to further improve code, for example by converting `"{}".format(var)` strings to `f{var}` strings. This particular example was determined to be out of scope of our formatter `Black` (see https://github.com/psf/black/issues/363)

**Comment by forsyth2** [comment]
This issue in particular came about because I like the `f{var}` strings better (they're more concise) but almost all the format-strings are in the other form. 

In general, there are different standards that are in use (e.g., PEP 8 -- https://peps.python.org/pep-0008/). We use `Black` (https://black.readthedocs.io/en/stable/) for formatting currently, but it doesn't cover the cases that `pyupgrade` does. 

I think it's less about defining what's acceptable and more about making things consistent (with some arbitrary standard) to improve code readability.

**Comment by xylar** [comment]
> I think it's less about defining what's acceptable and more about making things consistent (with some arbitrary standard) to improve code readability.

I think that’s totally right. It’s also about having tools (e.g. pycharm) that make it as easy as possible to learn the arbitrary conventions.

**Comment by xylar** [comment]
I’ve also really come to prefer f-strings so I support having a tool to make that move everywhere in zppy once and for all.

**Comment by golaz** [comment]
I agree that the formatting with f-strings is more elegant and intuitive than other options. But I do question the value of these mandates coming down from some external checker that we decided to use.

**Comment by forsyth2** [comment]
Well, `git grep -in "format(" | wc -l` returns `25`, so it'd tedious but not unreasonable to manually change them to f-strings.

**Comment by forsyth2** [comment]
(For comparison, `git grep -n "format(" | wc -l` return `357` in my E3SM Diags directory)

**Comment by xylar** [comment]
> But I do question the value of these mandates coming down from some external checker that we decided to use.

@golaz, I was also not onboard with linting code and following theses conventions for a few years when I started python coding.  But it's a bit like communication between fluent speakers of a language.  Someone used to the conventions can quickly and easily understand what someone else is saying (or the code they're writing), whereas someone not using the conventions takes quite a bit more effort to understand.

**Comment by forsyth2** [comment]
Closing, per https://github.com/E3SM-Project/zppy-interfaces/issues/4#issuecomment-3156580989
